Polishing assembly using raw read
Alignments of raw data with the hybrid assembly from SPAdes to improve the final assembly using Pilon
```
cd /fs/ess/scratch/PAS2303/Jonathan/Assembly/Polishing
mkdir Alignments
cd Alignments
mkdir Illumina Nanopore
```
Aligning Illumina reads using Bowtie2
```
module load bowtie2
module load samtools/1.6
# Making a Bowtie2 genome index
cd Illumina
bowtie2-build /fs/ess/scratch/PAS2303/Jonathan/Assembly/Joint_Assembly/SPAssembly/K55/scaffolds.fasta Illumina
# Aligning the raw data to the indexed genome. It generates a BAM file
bowtie2 -p 28 -x Illumina -1 /fs/ess/scratch/PAS2303/Jonathan/Assembly/Raw_Data/SRR2188474_1.fastq.gz -2 /fs/ess/scratch/PAS2303/Jonathan/Assembly/Raw_Data/SRR2188474_2.fastq.gz --very-sensitive | samtools sort > Illumina_bwt2.bam
# Indexing the BAM file
samtools index Illumina_bwt2.bam Illumina_bwt2.bai
```
Aligning Nanopore reads using BWA
```
cd ..; cd Nanopore
module load bwa/0.7.17
module load samtools/1.6
bwa index /fs/ess/scratch/PAS2303/Jonathan/Assembly/Joint_Assembly/SPAssembly/K55/scaffolds.fasta
bwa mem -t 28 /fs/ess/scratch/PAS2303/Jonathan/Assembly/Joint_Assembly/SPAssembly/K55/scaffolds.fasta /fs/ess/scratch/PAS2303/Jonathan/Assembly/Raw_Data/SRR11481813.fastq.gz | samtools sort -@8 -o Nanopore_bwa.bam -
# Indexing the BAM file
samtools index Nanopore_bwa.bam Nanopore_bwa.bai
```
Then, we proceed to polish the assembly with Pilon using the distinct alignments:
```
cd ..
# Illumina data only
java -Xmx125G -jar /fs/ess/scratch/PAS2303/Jonathan/Assembly/Software/pilon-1.23.jar --genome /fs/ess/scratch/PAS2303/Jonathan/Assembly/Joint_Assembly/SPAssembly/K55/scaffolds.fasta --frags /fs/ess/scratch/PAS2303/Jonathan/Assembly/Polishing/Alignments/Illumina/Illumina_bwt2.bam --output Polishing_Illumina --outdir /fs/ess/scratch/PAS2303/Jonathan/Assembly/Polishing --threads 28 --verbose
# Nanopore data only
java -Xmx125G -jar /fs/ess/scratch/PAS2303/Jonathan/Assembly/Software/pilon-1.23.jar --genome /fs/ess/scratch/PAS2303/Jonathan/Assembly/Joint_Assembly/SPAssembly/K55/scaffolds.fasta --bam /fs/ess/scratch/PAS2303/Jonathan/Assembly/Polishing/Alignments/Nanopore/Nanopore_bwa.bam --output Polishing_Nanopore --outdir /fs/ess/scratch/PAS2303/Jonathan/Assembly/Polishing --threads 28 --verbose
# Illumina + Nanopore
java -Xmx125G -jar /fs/ess/scratch/PAS2303/Jonathan/Assembly/Software/pilon-1.23.jar --genome /fs/ess/scratch/PAS2303/Jonathan/Assembly/Joint_Assembly/SPAssembly/K55/scaffolds.fasta --frags /fs/ess/scratch/PAS2303/Jonathan/Assembly/Polishing/Alignments/Illumina/Illumina_bwt2.bam --bam /fs/ess/scratch/PAS2303/Jonathan/Assembly/Polishing/Alignments/Nanopore/Nanopore_bwa.bam --output Polishing_Illumina+Nanopore --outdir /fs/ess/scratch/PAS2303/Jonathan/Assembly/Polishing --threads 28 --verbose
```
Comparisons
Assembly from SPades
```
/fs/ess/scratch/PAS2303/Jonathan/Assembly/Software/bbmap/stats.sh /fs/ess/scratch/PAS2303/Jonathan/Assembly/Joint_Assembly/SPAssembly/K55/scaffolds.fasta > Original.txt
```
First Pilon correction with Illumina bowtie2 bam
```
/fs/ess/scratch/PAS2303/Jonathan/Assembly/Software/bbmap/stats.sh /fs/ess/scratch/PAS2303/Jonathan/Assembly/Polishing/Polishing_Illumina > Polishing_Illumina.txt
```
Second Pilon correction with Nanopore bwa bam
```
/fs/ess/scratch/PAS2303/Jonathan/Assembly/Software/bbmap/stats.sh /fs/ess/scratch/PAS2303/Jonathan/Assembly/Polishing/Polishing_Nanopore.fasta > Polishing_Nanopore.txt
```
Third Pilon correction using both Illumina and Nanopore raw data bam files
```
/fs/ess/scratch/PAS2303/Jonathan/Assembly/Software/bbmap/stats.sh /fs/ess/scratch/PAS2303/Jonathan/Assembly/Polishing/Polishing_Illumina+Nanopore.fasta > Polishing_Illumina+Nanopore.txt
```
Which one does look better?
